modules/apps/27-interchain-accounts/controller/keeper/account.go

```diff
@@ -65,16 +64,7 @@ func (k Keeper) registerInterchainAccount(ctx context.Context, connectionID, por
 		return "", errorsmod.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s on connection %s", activeChannelID, portID, connectionID)
 	}
 
-	switch {
-	case k.portKeeper.IsBound(ctx, portID) && !k.hasCapability(ctx, portID):
```

Comment:
This case basically says - if a a capability exists (port is bound) with core ibc, but not claimed by the ica app, then return an error. This is essentially unreachable code, unless someone has wired a code path into the chain to allow icacontroller- port ids to be created with the core ibc scoped keeper.